src/main/scala/viper/gobra/translator/util/ViperWriter.scala

```diff
@@ -370,6 +370,42 @@ object ViperWriter {
       } yield ()
     }
 
+    /* Emits Viper statements. */
+    def assertWithDefaultReason(cond: vpr.Exp, error: Source.Verifier.Info => VerificationError): Writer[Unit] = {
+      val res = vpr.Assert(cond)(cond.pos, cond.info, cond.errT)
+      for {
+        _ <- write(res)
+        _ <- errorT({
+          case e@ vprerr.AssertFailed(Source(info), reason, _) if e causedBy res =>
+            error(info) dueTo DefaultErrorBackTranslator.defaultTranslate(reason)
+        })
+      } yield ()
+    }
+
+    /* Emits Viper statements. */
+    def exhale(cond: vpr.Exp, reasonT: (Source.Verifier.Info, ErrorReason) => VerificationError): Writer[Unit] = {
+      val res = vpr.Exhale(cond)(cond.pos, cond.info, cond.errT)
+      for {
+        _ <- write(res)
+        _ <- errorT({
+          case e@ vprerr.ExhaleFailed(Source(info), reason, _) if e causedBy res =>
+            reasonT(info, reason)
+        })
+      } yield ()
+    }
+
+    /* Emits Viper statements. */
+    def exhaleWithDefault(cond: vpr.Exp, error: Source.Verifier.Info => VerificationError): Writer[Unit] = {
```

Comment:
would rename it to `exhaleWithDefaultReason` to be consistent with `assertWithDefaultReason`